homeassistant/components/matter/api.py

```diff
@@ -173,3 +180,124 @@ async def websocket_set_wifi_credentials(
         ssid=msg["network_name"], credentials=msg["password"]
     )
     connection.send_result(msg[ID])
+
+
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "matter/node_diagnostics",
+        vol.Required(DEVICE_ID): str,
+    }
+)
+@websocket_api.async_response
+@async_handle_failed_command
+@async_get_matter_adapter
+async def websocket_node_diagnostics(
+    hass: HomeAssistant,
+    connection: ActiveConnection,
+    msg: dict[str, Any],
+    matter: MatterAdapter,
+) -> None:
+    """Gather diagnostics for the given node."""
+    node = node_from_ha_device_id(hass, msg[DEVICE_ID])
+    if not node:
+        raise MatterError("Node not found")
+    result = await matter.matter_client.node_diagnostics(node_id=node.node_id)
+    connection.send_result(msg[ID], dataclass_to_dict(result))
+
+
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "matter/ping_node",
+        vol.Required(DEVICE_ID): str,
+    }
+)
+@websocket_api.async_response
+@async_handle_failed_command
+@async_get_matter_adapter
+async def websocket_ping_node(
+    hass: HomeAssistant,
+    connection: ActiveConnection,
+    msg: dict[str, Any],
+    matter: MatterAdapter,
+) -> None:
+    """Ping node on the currently known IP-adress(es)."""
+    node = node_from_ha_device_id(hass, msg[DEVICE_ID])
+    if not node:
+        raise MatterError("Node not found")
+    result = await matter.matter_client.ping_node(node_id=node.node_id)
+    connection.send_result(msg[ID], result)
+
+
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "matter/open_commissioning_window",
+        vol.Required(DEVICE_ID): str,
+    }
+)
+@websocket_api.async_response
+@async_handle_failed_command
+@async_get_matter_adapter
+async def websocket_open_commissioning_window(
+    hass: HomeAssistant,
+    connection: ActiveConnection,
+    msg: dict[str, Any],
+    matter: MatterAdapter,
+) -> None:
+    """Open a commissioning window to commission a device present on this controller to another."""
+    node = node_from_ha_device_id(hass, msg[DEVICE_ID])
+    if not node:
+        raise MatterError("Node not found")
```

Comment:
Please add a test for missing node for each of these websocket commands (even if we move the code to a decorator).